src/Evolution/Systems/CurvedScalarWave/BoundaryConditions/Outflow.cpp

```diff
@@ -0,0 +1,79 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/CurvedScalarWave/BoundaryConditions/Outflow.hpp"
+
+#include <cstddef>
+#include <memory>
+#include <optional>
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Characteristics.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeString.hpp"
+
+namespace CurvedScalarWave::BoundaryConditions {
+
+template <size_t Dim>
+std::unique_ptr<domain::BoundaryConditions::BoundaryCondition>
+Outflow<Dim>::get_clone() const noexcept {
+  return std::make_unique<Outflow>(*this);
+}
+
+template <size_t Dim>
+void Outflow<Dim>::pup(PUP::er& p) {
+  BoundaryCondition<Dim>::pup(p);
+}
+
+template <size_t Dim>
+Outflow<Dim>::Outflow(CkMigrateMessage* const msg) noexcept
+    : BoundaryCondition<Dim>(msg) {}
+
+template <size_t Dim>
+std::optional<std::string> Outflow<Dim>::dg_outflow(
+    const std::optional<tnsr::I<DataVector, Dim, Frame::Inertial>>&
+        face_mesh_velocity,
+    const tnsr::i<DataVector, Dim>& normal_covector,
+    const tnsr::I<DataVector, Dim>& /*normal_vector*/,
+    const Scalar<DataVector>& gamma1, const Scalar<DataVector>& lapse,
+    const tnsr::I<DataVector, Dim>& shift) const noexcept {
+  tnsr::a<DataVector, 3, Frame::Inertial> char_speeds{lapse.size()};
+  characteristic_speeds(make_not_null(&char_speeds), gamma1, lapse, shift,
+                        normal_covector);
+
+  if (face_mesh_velocity.has_value()) {
+    const auto face_speed = dot_product(normal_covector, *face_mesh_velocity);
+    for (auto& char_speed : char_speeds) {
+      char_speed -= get(face_speed);
+    }
+  }
+  for (size_t i = 0; i < char_speeds.size(); ++i) {
+    if (min(char_speeds[i]) < 0.) {
+      return MakeString{}
+             << "Detected negative characteristic speed at boundary with "
+                "outflowing boundary conditions specified. The speed is "
+             << min(char_speeds[i]) << "for index " << i
```

Comment:
I think a space before the `for` so that this would read `0.1 for` instead of `0.1for`?